sql/select.go

```diff
@@ -58,24 +59,45 @@ func (p *planner) Select(n *parser.Select) (planNode, error) {
 	exprs := make([]parser.Expr, 0, len(n.Exprs))
 	columns := make([]string, 0, len(n.Exprs))
 	for _, e := range n.Exprs {
-		switch t := e.(type) {
-		case *parser.StarExpr:
+		// If a QualifiedName has a StarIndirection suffix we need to match the
+		// prefix of the qualified name to one of the tables in the query and
+		// then expand the "*" into a list of columns.
+		if qname, ok := e.Expr.(*parser.QualifiedName); ok && qname.IsStar() {
 			if desc == nil {
-				return nil, fmt.Errorf("* with no tables specified is not valid")
+				return nil, fmt.Errorf("\"%s\" with no tables specified is not valid", qname)
 			}
-			for _, col := range desc.Columns {
-				columns = append(columns, col.Name)
-				exprs = append(exprs, &parser.QualifiedName{Base: parser.Name(col.Name)})
+			if e.As != "" {
+				// TODO(pmattis): Test this error. THe parser refuses to parse "* AS
```

Comment:
This comment is actually detritus. While `SELECT * AS foo` doesn't parse, `SELECT bar.* AS foo` does.